### PR TITLE
Remove any cookies/state after resetting authentication of ExchangeServiceBase

### PR DESCRIPTION
**Pull request by MikeN123**
This prevents an old authentication from being reused by proxy firewalls like TMG.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@MikeN123__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

File: src/main/java/microsoft/exchange/webservices/data/ExchangeServiceBase.java [hunk 1]
```diff
@@ -195,7 +198,14 @@ private void initializeHttpClient() {
     httpClient = httpClientBuilder.build();
   }
 
-  @Override
+  private void initializeHttpContext() {
+    // Use an own cookie store, instead of the httpClient's global store, so cookies get reset on reinitialization
```
Comment (serious6) [inline src/main/java/microsoft/exchange/webservices/data/ExchangeServiceBase.java]: maybe this one could also be added as javadoc?

**Comment by serious6** [comment]
looks good thanks for the contribution.

File: src/main/java/microsoft/exchange/webservices/data/ExchangeServiceBase.java (hunk 1 above)
Comment (MikeN123) [inline src/main/java/microsoft/exchange/webservices/data/ExchangeServiceBase.java]: I moved it to a Javadoc comment.

**Pull request merged**